pytransform3d/urdf.py

```diff
@@ -108,11 +108,14 @@ def set_joint(self, joint_name, value):
                 np.hstack((axis, (value,))))
             joint2A = transform_from(
                 joint_rotation, np.zeros(3), strict_check=self.strict_check)
-        else:
-            assert joint_type == "prismatic"
+        elif joint_type == "prismatic":
             joint_offset = value * axis
             joint2A = transform_from(
                 np.eye(3), joint_offset, strict_check=self.strict_check)
+        else:
+            assert joint_type == "fixed"
+            print("WARNING: Fixed joint cannot be set")
```

Comment:
I would raise an error or print a warning here.

Reply:
Thanks for the quick feedback :). I originally thought to raise an error, but wanted to first see if this conflicts with the unit tests. Good to hear you have the same idea

Reply:
I didn't run the tests, but I see no serious problems. We would only need additional unit tests for the new feature.

Reply:
- existing tests do not give issues.

> Hi,
> 
> thanks for the contribution. Looks reasonable to me. The only concern that I have is that I don't know what `get_joint_limits` should return for this joint. Should it raise an error?

I have currently implemented this such that 'fixed' joints are hard-coded with limits (0, 0). The limits therefore enforce the "fixedness" of the joint. Also, this ensures that this joint cannot be changed by _set_joint_ because it will cap the input to these limits.

Reply:
I think I'd rather see a warning somewhere though. Because it doesn't make sense to change the joint angle of a fixed joint.

Reply:
found it, using _warnings_